mlir/test/mlir-opt/issue-74236.mlir

```diff
@@ -0,0 +1,10 @@
+// RUN: mlir-opt -split-input-file -verify-diagnostics %s -canonicalize
+
+llvm.func @malloc(i64) -> !llvm.ptr
+func.func @func2(%arg0: index, %arg1: memref<13x13xi64>, %arg2: index) {
+  %cst_7 = arith.constant dense<1526248407> : vector<1xi64>
+  %1 = llvm.mlir.constant(1 : index) : i64
```

Comment:
The fact that we allow the `llvm.mlir.constant` to materialize an integer with another type is a bug I believe (unfortunately this creeped up and is quite involved to fix now, I tried... :( )